inprocess/src/main/java/io/grpc/inprocess/InProcessChannelBuilder.java

```diff
@@ -225,9 +222,23 @@ public InProcessChannelBuilder propagateCauseWithStatus(boolean enable) {
     return this;
   }
 
+  /**
+   * Sets whether to include the provided messageSize or not and is propagated
+   * forward to InProcessTransport. This was added to not calculate messageSize
+   * if already provided while calling builder.
+   *
+   * @param assumedMessageSize length of InProcess transport's messageSize
+   * @return this
+   */
+  public InProcessChannelBuilder assumedMessageSize(long assumedMessageSize) {
+    checkArgument(assumedMessageSize >= 0, "assumedMessageSize must be >= 0");
```

Comment:
Document that the size should be non-negative? Alternatively, we could document that a negative value disables the assumption; we typically provide a way to undo previous calls, but it isn't likely all that important here.